lib/ModuleManager.cpp

```diff
@@ -904,6 +908,7 @@ QoreAbstractModule* QoreModuleManager::loadSeparatedModule(ExceptionSink& xsink,
         std::string fileCode = QoreDir::get_file_content(filePath);
         userModule->getProgram()->parsePending(fileCode.c_str(), feature, &xsink, &xsink);
     }
+    fileList->deref(&xsink);
```

Comment:
instead of using deref manually, use
`ReferenceHolder<QoreListNode> fileList =  ....`

@davidnich is there some easy know how, what pointers should be deleted, what should be wrapped into reference holders? it's so hard to keep track of it, you have to go in implementation of everything... maybe some info/intro from you could help?

Reply:
@sejvlond if the value has a protected destructor, then it must be dereferenced, otherwise it should be deleted normally

Reply:
and btw why isn't the value already returned as `ReferenceHolder<T>` - you would either have to except it as reference holder - creating or copying reference holders should be really cheap, or you would call `.release()` by yourself, but from the very declaration of the function it would be clear, that it is reference counted object, right?

Reply:
done